lib/internal/fs/promises.js

```diff
@@ -792,48 +794,72 @@ async function readdirRecursive(originalPath, options) {
       ),
     ],
   ];
-
+  function filter(absolutePath) {
+    const directory = absolutePath.slice(0, absolutePath.lastIndexOf(sep));
+    return absolutePath === originalPath ||
+      typeof options.recursiveFilter !== 'function' ||
+      (typeof options.recursiveFilter === 'function' && options.recursiveFilter({
```

Comment:
In `fsPromises.readdir()` version, can `recursiveFilter` be async?

Reply:
I do not see the need.  There are no events or external actions here that would hang the thread.  Forcing it to become async would push it into another callstack and there is a minor performance penalty for doing so.

Reply:
Perhaps. I wondered if, for example, user would want to run `await fsPromises.stat(dir)` inside filter function.